### PR TITLE
1890: Add method deployKeyTitles

### PR DESCRIPTION
**Pull request by zhaosongzs**
Add a new method called getExpiredDeployKeys in HostedRepository

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-1890](https://bugs.openjdk.org/browse/SKARA-1890): Add method deployKeyTitles (**Enhancement** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1508/head:pull/1508` \
`$ git checkout pull/1508`

Update a local copy of the PR: \
`$ git checkout pull/1508` \
`$ git pull https://git.openjdk.org/skara.git pull/1508/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1508`

View PR using the GUI difftool: \
`$ git pr show -t 1508`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1508.diff">https://git.openjdk.org/skara/pull/1508.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1508#issuecomment-1522404760)

### CODE REVIEW

**Comment by erikj79** [comment]
I'm not sure about the "Expired" part of the method name. In the forge, the keys aren't expired, we are just searching for keys older than a certain duration. Also note that the prefix 'get' isn't used much in Skara code. I would suggest `HostedRepository::deployKeyTitles`.

**Comment by zhaosongzs** [comment]
> I'm not sure about the "Expired" part of the method name. In the forge, the keys aren't expired, we are just searching for keys older than a certain duration. Also note that the prefix 'get' isn't used much in Skara code. I would suggest `HostedRepository::deployKeyTitles`.

Yes, makes sense. I will change the name of this method. Thx

File: forge/src/test/java/org/openjdk/skara/forge/github/GitHubRestApiTests.java [hunk 1]
```diff
@@ -268,4 +268,14 @@ void testDeleteDeployKey() {
         var githubRepo = githubRepoOpt.get();
         githubRepo.deleteDeployKeys(Duration.ofHours(24));
     }
+
+    @Test
+    void testGetExpiredDeployKeys() {
```
Comment (erikj79) [inline forge/src/test/java/org/openjdk/skara/forge/github/GitHubRestApiTests.java:273]: Update test name?

**Review by erikj79 — COMMENTED** [review]
Maybe also update the bug title? I would recommend not mentioning the exact method name in the title, but instead describing the new API. Something like "Add API to get deploy keys for a repository".

File: forge/src/test/java/org/openjdk/skara/forge/github/GitHubRestApiTests.java (hunk 1 above)
Comment (zhaosongzs) [reply forge/src/test/java/org/openjdk/skara/forge/github/GitHubRestApiTests.java:273]: Will fix it, thanks for catching it!

**Review by erikj79 — APPROVED** [review]

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**